File: crates/source-languages/tests/fixtures/sample.py
"""Catalog inventory tracking."""
import collections

DEFAULT_PAGE_SIZE = 25


class Catalog:
    """Tracks stock levels per SKU."""

    def __init__(self):
        self.items = collections.defaultdict(int)

    def add(self, sku, units):
        normalized = self.normalize(sku)
        self.items[normalized] += units
        return self.items[normalized]

    @staticmethod
    def normalize(sku):
        return sku.strip().upper()

    def total(self):
        return sum(self.items.values())


def build_catalog():
    catalog = Catalog()
    catalog.add("a-1", DEFAULT_PAGE_SIZE)
    return catalog

File: crates/source-languages/tests/fixtures/sample.lua
local DEFAULT_PAGE_SIZE = 25

local Catalog = {}
Catalog.__index = Catalog

function Catalog.new()
	return setmetatable({ items = {} }, Catalog)
end

local function normalize(sku)
	return string.upper(sku)
end

function Catalog:add(sku, units)
	local normalized = normalize(sku)
	self.items[normalized] = (self.items[normalized] or 0) + units
	return self.items[normalized]
end

function Catalog:restock(units)
	return self:add("DEFAULT", units)
end

local function build_catalog()
	local catalog = Catalog.new()
	catalog:add("a-1", DEFAULT_PAGE_SIZE)
	return catalog
end

return { Catalog = Catalog, build_catalog = build_catalog }
